compiler/rustc_parse/src/parser/tests.rs

```diff
@@ -1376,6 +1376,127 @@ fn ttdelim_span() {
     });
 }
 
+// Uses a macro rather than a function so that failure messages mention the
+// correct line in the test function.
```

Comment:
What about `#[track_caller]` + fn?

Reply:
I tried that, it didn't work :/

Reply:
Ah, right. Requires `#[track_caller]` on closures (requires features `stmt_expr_attributes`, `closure_track_caller`, #87417)

Reply:
Closure? I tried making `look` a top-level function like this:
```rust
#[track_caller]
fn look(p: &Parser<'_>, dist: usize, kind: rustc_ast::token::TokenKind) {
    p.look_ahead(dist, |tok| assert_eq!(kind, tok.kind));
}
```
and that's what didn't work. The assertion failure still reported the error on the line containing the `look_ahead` call. I'm not sure why it didn't work, it seems like an ideal use for `#[track_caller]`.

Reply:
I'm referring to

```rs
#![feature(closure_track_caller)]

#[track_caller]
fn look(p: &Parser<'_>, dist: usize, kind: TokenKind) {
    p.look_ahead(dist, #[track_caller] |tok| assert_eq!(kind, tok.kind));
}
```

but that also points inside `look_ahead` for some reason.